Use a TypedDict for plugin options config - closes #706

jaredkeil opened the pull request:
Implemented `PostgresqlConfigDict(TypedDict)` which has the same keys as the previous `config` dict. The value types are what is expected from the `request.config` object after .ini or command line options are parsed, and in general fit the expected usages of `config["<some_option>"]`. Now, when a variable is set by getting a config value, there is a type hint for that variable. For example:
```python
config = get_config(request)
port = config["port"]
# type hint -- port: str | None
```

fizyk commented:
@jaredkeil tests fail on python 3.7: 
* remove python 3.7 from the CI/package definition, and require at least 3.8
* Add newsfragments (break for dropping python 3.7 support and misc for introducing Typed config dict)
* adjust typing according to mypy issues

fizyk commented:
Add newsfragment `pipenv run towncrier create [issue_number].[type].rst`

jaredkeil commented:
Summary of changes which now pass the checks:
- Removed 3.7 from the workflow "Run tests" python-version matrix
- Updated setup.cfg `python_requires` at least 3.8, removed Python 3.7 lang from classifiers
- Resolved mypy issues by updating config types  and return type of `get_postgresql_option`
- Resolved black issue by adding trailing comma to `PostgresqlConfigDict` instantiation args
- Changed raw `list` type of `config["load"]` to `List` for older python version
- Added 2 newsfragments, one for break, one for misc, each linked to original issue #706.

fizyk commented:
Thank you @jaredkeil
